src/core/meshcop/border_agent.cpp

```diff
@@ -220,80 +198,86 @@ void BorderAgent::HandleCoapResponse(void *               aContext,
         }
     }
 
-    forwardContext.ToHeader(header, static_cast<Coap::Header *>(aHeader)->GetCode());
+    forwardContext.ToHeader(*message, response->GetCode());
 
-    if (message->GetLength() - message->GetOffset() > 0)
+    if (response->GetLength() - response->GetOffset() > 0)
     {
-        header.SetPayloadMarker();
+        message->SetPayloadMarker();
     }
 
-    SuccessOrExit(error = borderAgent.ForwardToCommissioner(header, *message));
+    SuccessOrExit(error = borderAgent.ForwardToCommissioner(*message, *response));
 
 exit:
-    if (error != OT_ERROR_NONE)
+    do
     {
+        if (error == OT_ERROR_NONE)
+        {
+            break;
+        }
+
         otLogWarnMeshCoP("Commissioner request[%hu] failed: %s", forwardContext.GetMessageId(),
-                         otThreadErrorToString(aResult));
-        forwardContext.ToHeader(header, CoapCodeFromError(error));
-        borderAgent.SendErrorMessage(header);
-    }
+                         otThreadErrorToString(error));
+
+        if (message == NULL)
```

Comment:
nit: keep a separate method for sending an error message to avoid this code style?